kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -367,9 +446,29 @@ private SnapshotImpl getSnapshotAtInit(Engine engine)
             getLogSegmentFrom(engine, lastCheckpointOpt);
 
         return logSegmentOpt
-            .map(logSegment -> createSnapshot(
-                logSegment, engine))
-            .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+                .map(logSegment -> getCoordinatedCommitsAwareSnapshot(engine, logSegment))
+                .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+    }
+
+    private SnapshotImpl getCoordinatedCommitsAwareSnapshot(
+            Engine engine, LogSegment initialSegmentForNewSnapshot) {
+        SnapshotImpl newSnapshot = createSnapshot(initialSegmentForNewSnapshot, engine);
```

Comment:
Can you check if this breaks the snapshot hint optimization we have to avoid reloading the Protocol and Metadata for each get latest snapshot call?

Reply:
Thanks for pointing this out! It will break the snapshot hint optimization. Created issue #3437 for this.

Reply:
Add the issue as a comment here and specify this can be optimized.